software/28.Gin/README.md

```diff
@@ -0,0 +1,139 @@
+# Workshop 2 - API REST in Go with Gin
+
+In this workshop we will learn how to use the main functions of the Gin package in Go and how to create a REST API.
+
+> :bulb: A quick [documentation](https://searchapparchitecture.techtarget.com/definition/RESTful-API) about REST API.
+
+## Step 0: Initialization
+
+All the required information to install the workshop's dependencies are given in the [SETUP.md](./SETUP.md)
+
+## Step 1: The codebase
+
+We designed a boilerplate for this project, it will be useful to start this project, but also to summarize all the required parts of an API
+We will have :
+
+- `controllers`: this is where you will design your routes endpoint. it will often be a wrapper that calls other API or the database itself
+- `middlewares`: those are functions that can be called to print logs ,check access rights before entering a controller or define a default endpoint
+- `routes`: the core of the router, setting up all the routes handler, with their middlewares and controllers
+
+There are plenty of other important package you may need in a real API, but these are the main one you can be sure you'll need one time or another. Now let's code.
+
+Add a route on the endpoint `/`:
+- It uses the `GET` method
+- It responds `"hello: world"` in a json format
+
+> You can test it with `http://localhost:8080/` in a browser or with `postman`
+
+## Step 2: Let's get started
+
+You are now fine with the creation of endpoint.
+Next step is to create different request method.
+There is many [request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), you already experimented the `GET` request.
+
+For this step, we will use some others' method:
+- a `GET` request `/item/template`: to get the template of the item as follows.
+```json
+{
+  "name": "name",
+  "description": "description",
+  "price": 0
+}
+```
+
+- a `POST` request `/item/display`: to display the item passed in the body as follows.
+  <img src="./.assets/request-item-display.png">
+
+
+also try to implement a default endpoint for the URL's that doesn't match any routes. (404 notfound error)
+
+
```

Comment:
It's very important two indicates where we need to work (you need to add this in all steps).